### ISSUE TITLE
IDE crash

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#1595]
Code to reproduce:
```cs
static rettype(T) Invoke<T>(T fn, params T args) where T : Delegate
{
    fn(
}
```
Tested with: Nightly 06/16/2022

**Comment by Fusioon** [comment]
```cs
class Program
{
    static rettype(T) Invoke<T>(T fn, params T args) where T : Delegate
    {
        var result = fn(params args);
        return result;
    }


    static void Foo(int result)
    {

    }

    public static void Main()
    {
        Invoke<delegate void(int result)>(scope => Foo, 10);
    }
}
```
Also this code fails to compile because delegate parameter name is clashing with variable inside `Invoke<T>` is this intended behaviour or bug?

**Comment by bfiete** [comment]
First fixed at 1b3b693cb2f540dbb199f69cdc656f2e227880a2, second fixed at 67b30b639dbc38ad01200e15b1acba222a04ee17

**Comment by bfiete** [comment]
(but, yes - the name clashing was a bug)
